TUnit.Assertions/Compare.cs

```diff
@@ -106,25 +117,26 @@ public static IEnumerable<ComparisonFailure> CheckEquivalent<T>(T actual, T expe
                 yield break;
             }
 
-            foreach (var comparisonFailure in CheckEquivalent(actualFieldValue, expectedFieldValue, options, [..memberNames, fieldInfo.Name], MemberType.Field))
+            foreach (var comparisonFailure in CheckEquivalent(actualFieldValue, expectedFieldValue, options, [..memberNames, fieldName], MemberType.Field))
             {
                 yield return comparisonFailure;
             }
         }
         
-        foreach (var propertyInfo in actual.GetType().GetProperties().Concat(expected!.GetType().GetProperties())
+        foreach (var propertyName in actual.GetType().GetProperties().Concat(expected.GetType().GetProperties())
                      .Distinct()
-                     .Where(p => p.GetIndexParameters().Length == 0))
+                     .Where(p => p.GetIndexParameters().Length == 0)
+                     .Select(x => x.Name))
```

Comment:
Should this `Distinct` also be moved to the end here?

Reply:
#1220